specs/_template/01_main.rst

```diff
@@ -0,0 +1,253 @@
+-----------------------
+
+You can put general information or context for all use case scenarios here.
+Feel free to remove the use case explanation below.
+
+Every use case scenario consists of preconditions – prerequisites
+and assumptions, steps - numbered sequence of steps of the scenario,
+from the user's perspective, and postconditions - results and
+assumptions that must hold after the steps finished.
+
+A use case can *inherit* another use case. That means that
+inheriting use case takes all of the preconditions, steps, and postconditions
+from the inhertied use case.
+
+The inheriting (child) use case
+can override (replace) any step or precondition in the parent
+use case. The overriden steps in child should match the step number
+in parent. The child use case may introduce sub-steps, in that case
+numbering is starts with dot, for example "3.1", "3.2" gives more
+details about step "3" of a parent use case.
+
+4.1. Happy Case
+~~~~~~~~~~~~~~~~~
+
+.. _happy_case_preconditions:
+
+Preconditions
++++++++++++++
+
+* Things that are expected to be true and required for the steps below
+* This are the things that needed for the functionality to work properly
+
+.. _happy_case_steps:
+
+Steps
++++++
+
+1. <Open something>
+
+2. Select "X" action. Something opens.
+
+.. you can reference some steps in other scenarios, when needed:
+
+.. _happy_case_step_3:
+
+3. Type in "abcd". and so on
+
+
+.. _happy_case_postconditions:
+
+Postconditions
+++++++++++++++
+
+* Changes to the database, to user interface
+* Changes to the backend systems' databases
+* Changes in the smart contracts and so on.
+
+
+4.2. Edge Case Scenario
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Inherits from the `4.1. Happy Case`_
+
+.. Inherits from
+    means that this scenario takes all the preconditions,
+    steps and postconditions from another use case and possibly extends
+    or overrides them with new values.
+
+Preconditions
++++++++++++++
+
+.. if preconditions are all the same - remove this section completely.
+.. if preconditions are the same AND there's something additional, or something is missing
+.. then explicitly mention all preconditions as wel.
+
+* Precondition text
+
+Steps
++++++
+
+.. step number overrides the step with the same number in the parent (inherited from) scenario.
+
+3. 1. Do something else instead. Something else happening.
+
+Postconditions
+++++++++++++++
+
+.. delete the postconditions section if they are the same as in parent scenario
+.. otherwise, list all postconditions here.
+
+* Some other postcondition
+
+
+4.3. Remember loading state scenario
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Inherits from the `4.1. Happy Case`_
+
+Steps
++++++
+
+2. Something is loading
+
+4.5. Remember error situations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Inherits from the `4.1. Happy Case`_
+
+Steps
++++++
+
+3. Error happens. Display error to the user
+
+
+.. _`iOS`: 02_user_interface_ios.rst
+.. _`Android`: 02_user_interface_android.rst
```

Comment:
I would not necessarly separate all user interfaces into different files. 

Actually I would prefer the most if they would be directly part of the use cases so that you can directly see the related screens to a story. But that is only really possible if we can somehow hide/collapse that info in rst

Reply:
I still think it's nice if we would have iOS and Android side by side.